app/controllers/incomeSources/add/IncomeSourcesAccountingMethodController.scala

```diff
@@ -104,41 +107,56 @@ class IncomeSourcesAccountingMethodController @Inject()(val authenticate: Authen
     )(user, messages)))
   }
 
+  def convertToIncomeSourceType(incomeSource: String): Either[Throwable, IncomeSourceType] = {
+    incomeSource match{
+      case SelfEmployment.key => Right(SelfEmployment)
+      case UkProperty.key => Right(SelfEmployment)
+      case ForeignProperty.key => Right(ForeignProperty)
+      case _ => Left(new Error)
```

Comment:
if you can provide error details in the error, as this method contract against "external world" would be something like: 
  Success Or Failed because of ...